ssl/record/ssl3_buffer.c

```diff
@@ -58,6 +58,11 @@ int ssl3_setup_read_buffer(SSL *s)
         if (ssl_allow_compression(s))
             len += SSL3_RT_MAX_COMPRESSED_OVERHEAD;
 #endif
+
+        /* Ensure our buffer is large enough to support all our pipelines */
+        if (s->max_pipelines > 1)
```

Comment:
Drop the if and do the multiplication always?

A branch mispredict will be much more expensive than the multiply.
It seems like max_pipelines should always be ≥ 1 but there are places in the code that don't appear to enforce this.

There is another instance of this check in tls_common.c

Reply:
Unfortunately `max_pipelines` defaults to 0 meaning "don't use pipelining" - so this doesn't work.